src/MongoDB/Manager.c

```diff
@@ -301,7 +301,70 @@ static PHP_METHOD(Manager, executeCommand)
 
 	intern = Z_MANAGER_OBJ_P(getThis());
 
-	phongo_execute_command(intern->client, db, command, options, -1, return_value, return_value_used TSRMLS_CC);
+	phongo_execute_command(intern->client, PHONGO_COMMAND_RAW, db, command, options, -1, return_value, return_value_used TSRMLS_CC);
+} /* }}} */
+
+/* {{{ proto MongoDB\Driver\Cursor MongoDB\Driver\Manager::executeReadCommand(string $db, MongoDB\Driver\Command $command[, array $options = null])
+   Execute a ReadCommand */
+static PHP_METHOD(Manager, executeReadCommand)
+{
+	php_phongo_manager_t     *intern;
+	char                     *db;
+	phongo_zpp_char_len       db_len;
+	zval                     *command;
+	zval                     *options = NULL;
+	DECLARE_RETURN_VALUE_USED
+	SUPPRESS_UNUSED_WARNING(return_value_ptr)
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "sO|z!", &db, &db_len, &command, php_phongo_command_ce, &options) == FAILURE) {
+		return;
+	}
+
+	intern = Z_MANAGER_OBJ_P(getThis());
+
+	phongo_execute_command(intern->client, PHONGO_COMMAND_READ, db, command, options, -1, return_value, return_value_used TSRMLS_CC);
+} /* }}} */
+
+/* {{{ proto MongoDB\Driver\Cursor MongoDB\Driver\Manager::executeWriteCommand(string $db, MongoDB\Driver\Command $command[, array $options = null])
+   Execute a WriteCommand */
+static PHP_METHOD(Manager, executeWriteCommand)
+{
+	php_phongo_manager_t     *intern;
+	char                     *db;
+	phongo_zpp_char_len       db_len;
+	zval                     *command;
+	zval                     *options = NULL;
+	DECLARE_RETURN_VALUE_USED
+	SUPPRESS_UNUSED_WARNING(return_value_ptr)
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "sO|z!", &db, &db_len, &command, php_phongo_command_ce, &options) == FAILURE) {
```

Comment:
Are we inconsistent about using `zend_replace_error_handling()`, or was that intentionally only done to ensure we throw exceptions from constructors? This is a general question outside the scope of this PR.

Reply:
I think it was intentionally. There is no other way to signal in Constructors that ctor options were wrong.